core/src/main/java/hudson/cli/InstallPluginCommand.java

```diff
@@ -172,11 +167,20 @@ protected int run() throws Exception {
         return 0; // all success
     }
 
-    private static FilePath getTargetFilePath(String name) {
-        return new FilePath(getTargetFile(name));
+    private static File getTmpFile() throws Exception {
+        return File.createTempFile("download", ".jpi.tmp", Jenkins.get().getPluginManager().rootDir);
     }
 
-    private static File getTargetFile(String name) {
-        return new File(Jenkins.getActiveInstance().getPluginManager().rootDir,name+".jpi");
+    private static File moveToFinalLocation(File tmpFile) throws Exception {
+        String pluginName;
+        try (JarFile jf = new JarFile(tmpFile)) {
+            pluginName = jf.getManifest().getMainAttributes().getValue("Short-Name");
```

Comment:
Can lead to NPEs for corrupted files, e.g. when there is no manifest
https://docs.oracle.com/javase/7/docs/api/java/util/jar/JarFile.html#getManifest()

Reply:
So…the command will fail, as it should. Not an actual problem.